src/bp/core/routers/admin/users.ts

```diff
@@ -45,34 +46,30 @@ export class UsersRouter implements CustomRouter {
           })
         )
         const email = req.body.email
-        const alreadyExists = await this.authService.findUserByEmail(email, ['id'])
+        const alreadyExists = await this.authService.findUserByEmail(email, ['email'])
 
         if (alreadyExists) {
           throw new InvalidOperationError(`User ${email} is already taken`)
         }
 
-        const tempPassword = await this.authService.createUser(email)
+        const createdUser: CreatedUser = await this.authService.createUser({ email })
 
         return sendSuccess(res, 'User created successfully', {
           email,
-          tempPassword
+          tempPassword: createdUser.password
         })
       })
     )
 
     router.delete(
-      '/:userId', // Delete user
+      '/:email', // Delete user
```

Comment:
IMO it's a little weird to pass the user's email as req params, but since we don't have id, there's no better option ... except if we stil have an id that is the email's hash ...

Reply:
an email should be params-friendly, but anything could be used instead if required